File: src/main/java/com/imagegame/service/ItemStoreService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.ItemStore;
import com.imagegame.dto.ItemStoreDTO;
import com.imagegame.dto.ItemStoreSearchDTO;
import com.imagegame.dto.ItemStorePageDTO;
import com.imagegame.dto.ItemStoreConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ItemStoreService extends GenericService<ItemStore, Integer> {

	List<ItemStore> findAll();

	ResultDTO addItemStore(ItemStoreDTO itemStoreDTO, RequestDTO requestDTO);

	ResultDTO updateItemStore(ItemStoreDTO itemStoreDTO, RequestDTO requestDTO);

    Page<ItemStore> getAllItemStores(Pageable pageable);

    Page<ItemStore> getAllItemStores(Specification<ItemStore> spec, Pageable pageable);

	ResponseEntity<ItemStorePageDTO> getItemStores(ItemStoreSearchDTO itemStoreSearchDTO);
	
	List<ItemStoreDTO> convertItemStoresToItemStoreDTOs(List<ItemStore> itemStores, ItemStoreConvertCriteriaDTO convertCriteria);

	ItemStoreDTO getItemStoreDTOById(Integer itemStoreId);







}

File: imagegame_ui/src/services/ItemStoreService.js
import http from "../http-common"; 

class ItemStoreService {
  getAllItemStores(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/itemStore/itemStores`, searchDTO);
  }

  get(itemStoreId) {
    return this.getRequest(`/itemStore/${itemStoreId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/itemStore?field=${matchData}`, null);
  }

  addItemStore(data) {
    return http.post("/itemStore/addItemStore", data);
  }

  update(data) {
  	return http.post("/itemStore/updateItemStore", data);
  }
  
  uploadImage(data,itemStoreId) {
  	return http.postForm("/itemStore/uploadImage/"+itemStoreId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ItemStoreService();
